lib/Query_Cache.cpp

```diff
@@ -331,8 +412,37 @@ Query_Cache::Query_Cache() {
 	purge_threshold_pct_min=DEFAULT_purge_threshold_pct_min;
 	purge_threshold_pct_max=DEFAULT_purge_threshold_pct_max;
 	max_memory_size=DEFAULT_SQC_size;
+
+	// Initialize prometheus metrics
+	init_prometheus_counter_array<qc_metrics_map_idx, p_qc_counter>(qc_metrics_map, this->metrics.p_counter_array);
+	init_prometheus_gauge_array<qc_metrics_map_idx, p_qc_gauge>(qc_metrics_map, this->metrics.p_gauge_array);
 };
 
+void Query_Cache::p_update_metrics() {
```

Comment:
See my comment on https://github.com/sysown/proxysql/pull/2676/files#r411761545 .
Let's simplify the code

Reply:
Simplified!